### PR TITLE
feat: Added Smart substitution support for S3

### PR DESCRIPTION
**Pull request by nidhi-nair**
## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- JUnit test case
- Manual testing

### CODE REVIEW

**Review by sumitsum — APPROVED** [review]

File: app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java [hunk 1]
```diff
@@ -283,58 +289,116 @@ String readFile(AmazonS3 connection, String bucketName, String path, Boolean enc
...
+            } else if (smartSubstitutionObject instanceof String) {
+                // Older UI configuration used to set this value as a string which may/may not be castable to a boolean
+                // directly. This is to ensure we are backward compatible
+                smartBsonSubstitution = Boolean.parseBoolean((String) smartSubstitutionObject);
+            }
+
+            try {
+                // Smartly substitute in Json fields and replace all the bindings with values.
+                if (TRUE.equals(smartBsonSubstitution)) {
+                    final String body = actionConfiguration.getBody() != null ? actionConfiguration.getBody() : "";
```
Comment (trishaanand) [inline app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java:322]: Is body the only JSON field for S3?

Comment (nidhi-nair) [reply app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java:322]: Yes the others are string values for configs or file names

**Review by trishaanand — APPROVED** [review]

File: app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java [hunk 2]
```diff
@@ -283,58 +289,116 @@ String readFile(AmazonS3 connection, String bucketName, String path, Boolean enc
...
+        public Mono<ActionExecutionResult> executeParameterized(AmazonS3 connection,
+                                                                ExecuteActionDTO executeActionDTO,
+                                                                DatasourceConfiguration datasourceConfiguration,
+                                                                ActionConfiguration actionConfiguration) {
+
+
+            final Map<String, Object> formData = actionConfiguration.getFormData();
+            List<Map.Entry<String, String>> parameters = new ArrayList<>();
+
+            Boolean smartBsonSubstitution = TRUE;
```
Comment (trishaanand) [inline app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java:307]:
```suggestion
            Boolean smartJsonSubstitution = TRUE;
```

**Review by sumitsum — APPROVED** [review]

**Review by sumitsum — APPROVED** [review]

**Pull request merged**